### ISSUE TITLE
`no-use-before-define` rule: allow mutual call for arrow functions

### ISSUE THREAD

**Issue by ollazarev** [eslint/eslint#9861]
1. ESLint version is 4.14.0
1. `no-use-before-define` rule
1. In such cases, when there is a mutual call in arrow functions, the following error occurs: `func2 was used before it was defined`.
```javascript
let func1 = () => {
  setTimeout(func2, 1000);
};
let func2 = () => {
  setTimeout(func1, 1000);
};
func1();
```
4. In this case, the error should not occur, because the functions are declared sequentially, and their call occurs after they are declared.
4. I think that it is necessary to add a new option to `no-use-before-define` rule, which will avoid the occurrence of this error.

**Comment by not-an-aardvark** [comment]
Does the [`functions: false`](https://eslint.org/docs/rules/no-use-before-define#options) option solve this problem?

**Comment by ollazarev** [comment]
No.
I checked with the following settings:
```javascript
{
    "no-use-before-define": ["error", { "functions": false }]
}
```

**Comment by not-an-aardvark** [comment]
Oops, I think I meant to ask about `variables: false`. (In hindsight the option names are a bit unintuitive.)

**Comment by ollazarev** [comment]
The issue is resolved, thank you.
